src/program.cpp

```diff
@@ -2159,13 +2162,29 @@ cvk_entry_point::create_pipeline(const cvk_spec_constant_map& spec_constants) {
         specConstantData.data(),
     };
 
+    void* pipelineShaderStageCreateInfoPNext = nullptr;
+    VkPipelineShaderStageRequiredSubgroupSizeCreateInfo
+        reqdSubgroupSizeCreateInfo;
+    if (m_device->supports_controled_subgroups()) {
+        auto reqdSubgroupSize = m_program->required_sub_group_size(m_name);
+        if (reqdSubgroupSize > m_device->max_sub_group_size() ||
+            reqdSubgroupSize < m_device->min_sub_group_size()) {
```

Comment:
Shouldn't we fail in that case? We should not magically change the subgroup size kernel writers assumed would be used.